### ISSUE TITLE
using vector `class` argument in `expect_is` causes errors in `test_dir`

### ISSUE THREAD

**Issue by goldingn** [hadley/testthat#293]
##### TL;DR: When calling `test_dir` on a test containing an `expect_is` call with character vector `class` argument, two errors are reported, causing an error in a `vapply` call before ever reporting the failures.

`expect_is` mangles the error message slightly when the `class` argument is a vector:

```r
x <- NA
class(x) <- c('a')
expect_is(x, c('b', 'c'))
```
`Error: x inherits from a not bx inherits from a not c`

which isn't so bad, but when used in test_dir (and quite possibly others) it leads to the following cryptic error:

```r
dir <- tempdir()
write("test_that('blah',{
         x <- NA
         class(x) <- c('a')
         expect_is(x, c('b', 'c'))
       })",
      file = paste0(dir, '/test-demo.R'))
test_dir(dir)
```
`Error in vapply(seq_len(n), function(i) { : values must be length 1, but FUN(X[[1]]) result is length 2`

The error is thrown by the following line in `test_dir`:
```r
current_reporter$end_reporter()
```

because `failure_summary` returns a vector of length two mangling the `vapply` call

```
[1] "1. Failure (at test-demo.R#4): blah -----------------------------------------\nx inherits from a not b"
[2] "1. Failure (at test-demo.R#4): blah -----------------------------------------\nx inherits from a not c"
```

Hope this is helpful!

**Comment by goldingn** [comment]
:+1:
